### PR TITLE
Update Ruby to vets-api's Ruby version #19606

### PR DESCRIPTION
**Pull request by zurbergram**
## Description

https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19606
https://github.com/department-of-veterans-affairs/vets.gov-team/issues/19612

## Testing
- [ ] Smoke test GIDS application and usage of GIDS endpoints

### CODE REVIEW

**Review by shawkey-daniel — APPROVED** [review]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
-      useradd -r -g gibct gibct
-
-RUN yum install -y git make gcc-c++ openssl-devel readline-devel zlib-devel sqlite-devel postgresql-devel socat timeout epel-release nc
-
-# Install Red Hat SCI library for ruby packages
-RUN yum install -y centos-release-scl-rh
-RUN yum install -y rh-ruby23 rh-ruby23-ruby-devel rh-ruby23-rubygems
-RUN echo "source /opt/rh/rh-ruby23/enable" > /etc/profile.d/rh-ruby23.sh
-RUN ["/bin/bash", "--login", "-c", "gem install --no-doc bundler"]
+    useradd -r -g gibct gibct && \
```
Comment (annaswims) [inline Dockerfile]: Please review this list and only add what is needed for gids.

File: Gemfile [hunk 2]
```diff
@@ -84,6 +84,8 @@ group :development, :test do
   gem 'guard-rspec', require: false
 
   gem 'factory_bot_rails'
+  # TODO: switch to a version number once that version is released
```
Comment (annaswims) [inline Gemfile]: I think you'd be better off with the released version.  This sha is from 2017.  Also changes to the gemfile should be done in a separate PR from the ruby upgrade.

Comment (zurbergram) [reply Gemfile]: Please see https://github.com/department-of-veterans-affairs/gibct-data-service/pull/429 as the plan is to merge that PR before this one

File: Dockerfile (hunk 1 above)
Comment (zurbergram) [reply Dockerfile]: - `build-essential` normal package to install on Debian to allow stuff to happen
- `git` needed to install sniffybara gem and others with specific sha refs
- `curl` needed to install [phantomjs](https://gist.github.com/julionc/7476620) (not the exact same process as listed)
- `clamav` needed to run [freshclam](https://wiki.archlinux.org/index.php/ClamAV)
- `libpq-dev` need to install gem `pg`
- `libgmp-dev` not needed, Multiprecision arithmetic library
- `imagemagick` not needed, pdf related
- `pdftk` not needed, pdf related
- `poppler-utils` not needed, pdf related

File: app/models/complaint.rb [hunk 3]
```diff
@@ -97,7 +97,9 @@ def ok_to_sum?
   end
 
   def set_facility_code_complaint
+    # rubocop:disable Performance/RegexpMatch
     COMPLAINT_COLUMNS.each_pair { |complaint, issue_regex| self[complaint] = issues =~ issue_regex ? 1 : 0 }
```
Comment (omgitsbillryan) [inline app/models/complaint.rb]: Why the exemption?

https://www.rubydoc.info/github/rubocop-hq/rubocop/master/RuboCop/Cop/Performance/RegexpMatch

After reading ☝️ seems like you want to just use `issues.match?(issue_regex)` - or am I missing something?

**Review by omgitsbillryan — CHANGES_REQUESTED** [review]

File: app/models/complaint.rb (hunk 3 above)
Comment (zurbergram) [reply app/models/complaint.rb]: Plan is to circle back and fix all of the `=~` in future work

Comment (omgitsbillryan) [reply app/models/complaint.rb]: Were there other `=~` errors caught? I dont see any in this PR.

Comment (zurbergram) [reply app/models/complaint.rb]: There are 6 usages of `=~` throughout the repo, not all of them were caught by rubocop

Comment (omgitsbillryan) [reply app/models/complaint.rb]: Is rubocop is misconfigured? 😬 or is missing these other usages intentional? Maybe a separate bug is warranted.

I'd vote for just fixing in this case becayse "future work" == "never".

Comment (zurbergram) [reply app/models/complaint.rb]: changed

**Review by omgitsbillryan — APPROVED** [review]
LGTM

**Comment by jholton** [comment]
@omgitsbillryan is the Dockerfile used by deployed environments?  Or is all that (e.g. Ruby version) handled by config in `devops`?

**Review by sonntag-adam — APPROVED** [review]
